src/coord/src/coord.rs

```diff
@@ -2526,7 +2527,13 @@ impl Coordinator {
             // reference could be caused by a user specifying an object in a different
             // schema than the first query. An index could be caused by a CREATE INDEX
             // after the transaction started.
-            if let Some(id) = stmt_ids.difference(&read_txn.timedomain_ids).next() {
+            //
+            // The call to `sorted` ensures the error message is deterministic.
```

Comment:
nit: We do already have a call to sort on `names` for the same reason. We could move that into a `sorted()` call as well and update the comment to say:

```
From 0bd3f59757e1d6fdfaf5a824abbf72b132582acf Mon Sep 17 00:00:00 2001
From: Ufuk Celebi <1756620+uce@users.noreply.github.com>
Date: Mon, 20 Sep 2021 16:04:19 +0200
Subject: [PATCH] nit

---
 src/coord/src/coord.rs | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/coord/src/coord.rs b/src/coord/src/coord.rs
index 4bb2e90b8..54abeaaea 100644
--- a/src/coord/src/coord.rs
+++ b/src/coord/src/coord.rs
@@ -2528,21 +2528,20 @@ impl Coordinator {
             // schema than the first query. An index could be caused by a CREATE INDEX
             // after the transaction started.
             //
-            // The call to `sorted` ensures the error message is deterministic.
+            // The calls to `sorted` ensure the error message is deterministic.
             if let Some(id) = stmt_ids
                 .difference(&read_txn.timedomain_ids)
                 .sorted()
                 .next()
             {
-                let mut names: Vec<_> = read_txn
+                let names: Vec<_> = read_txn
                     .timedomain_ids
                     .iter()
                     // This could filter out a view that has been replaced in another transaction.
                     .filter_map(|id| self.catalog.try_get_by_id(*id))
                     .map(|item| item.name().to_string())
+                    .sorted()
                     .collect();
-                // Sort so error messages are deterministic.
-                names.sort();
                 return Err(CoordError::RelationOutsideTimeDomain {
                     relation: self.catalog.get_by_id(&id).name().to_string(),
                     names,
--
2.25.1
```